src/hardware/link-mso19/api.c

```diff
@@ -140,8 +141,12 @@ static void mso_update_trigger_slope(struct dev_context *devc)
 static void mso_limit_trigger_level(struct dev_context *devc)
 {
 	double max_level = 2.0 * devc->dso_probe_factor;
-	if (devc->dso_trigger_level < -max_level) {
-		devc->dso_trigger_adjusted = -max_level;
+	double min_level = -max_level;
+	max_level -= devc->dso_offset_adjusted;
+	min_level -= devc->dso_offset_adjusted;
```

Comment:
isn't this going to set min_level to something outside of the interval [-max, +max] ?

Reply:
This code is correct. Think of it like a sliding window. the MSO can slide the window up or down based on DAC value.

So normally, you might do a [min, max] of [-2.0, 2.0], but you can also do [-1.0, 3.0] or [0.0, 4.0]

Reply:
Ok